src/main/scala/uk/gov/nationalarchives/tdr/api/auth/AuthorisationTags.scala

```diff
@@ -56,8 +56,8 @@ object ValidateSeries extends AuthorisationTag {
 
     bodyResult.map(body => {
       body.tdrCode match {
-        case Some(code) if code == userBody => continue
-        case Some(code) =>
+        case code if code == userBody => continue
+        case code if code != userBody =>
```

Comment:
Is this `if` statement necessary? Won't it work like before if it's not there?

Reply:
Because `String` can only be a `String` unlike `Option` which can be `Some` or `None`, line 60 will catch anything that isn't matched by line 59. This makes the `case _ => throw new IllegalStateException("")` redundant because you can never get there.
What I've realised now is that field is now mandatory so it never won't be populated, so we don't need the `case _ => throw new IllegalStateException("")` case and so we don't need the new if statement either. I've taken them both out.